src/main/java/hudson/plugins/templateproject/ProxyBuildEnvironment.java

```diff
@@ -0,0 +1,160 @@
+package hudson.plugins.templateproject;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.matrix.MatrixProject;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.Action;
+import hudson.model.BuildListener;
+import jenkins.model.DependencyDeclarer;
+import hudson.model.DependencyGraph;
+import hudson.model.Hudson;
+import hudson.model.Item;
+import hudson.model.Project;
+import hudson.model.Run;
+import hudson.security.AccessControlled;
+import hudson.tasks.BuildWrapper;
+import hudson.tasks.BuildWrapperDescriptor;
+import hudson.tasks.Messages;
+import hudson.util.FormValidation;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+public class ProxyBuildEnvironment extends BuildWrapper implements DependencyDeclarer {
+
+  private final String projectName;
```

Comment:
Although spaces might be more ideal, would be nice to have consistency amongst files ... which currently use tabs.